src/guide/advanced/intro.md

```diff
@@ -161,21 +158,21 @@ opt-level = "z"     # Optimize for size vs speed with "s"/"z"(removes vectorizat
 codegen-units = 1   # Further reduces binary size but increases compilation time
 ```
 
-Here we note a few things; firstly, when you encounter a panic in Rust, it stores a bit of debug information, so that you can interpret what happened inside the program, and debug, even when compiled in `release` mode. We explicitly state that we don't want any of that, and instead want panics to just stop the execution. We're not losing any functionality, despite what you might think, failure to execute a WASM blob isn't considered a warning-level event and is not logged even on the peer, so even if you had some useful information in the panic message, you have no reliable way of retrieving it.
+A few things should be noted; first, Rust stores a lot of debug information(even when compiled in `release` mode) to aid you when you encounter a panic, so that you can debug your application. All this debug information bloats the binary significantly, so instead of storing it, panics are directed to immediately abort the execution. By doing this no crucial functionality is lost, despite to what you might think. Panics are a rare event in the Rust world and the reason of failure to execute a WASM blob isn't logged on the peer, so even if you had some useful information in the panic message, you have no reliable way of retrieving it.
```

Comment:
```suggestion
Rust stores a lot of debug information (even when compiled in `release` mode), which is (as the name suggests)  used for debugging a panic in your Rust application. As you would expect, this information increases the size of the compiled WASM significantly. Normally, this would be a worthwhile trade-off, but not in our case. Firstly, since the WASM is stored on-chain, it will be permanently recorded in some block and take space on every full node of an Iroha network (which by the way, stores a lot of its information in RAM), so storage space for WASM is at a premium. Secondly, once the WASM smart contract is stored on-chain, the debug information is no longer accessible. Indeed, you shouldn't debug on a peer. Instead, you should replicate the conditions that caused the panic locally and debug on your personal machine. 
```